exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/DrillRoot.java

```diff
@@ -85,19 +86,24 @@ public ClusterInfo getClusterInfoJSON() {
     // For all other cases the user info need-not or should-not be displayed
     OptionManager optionManager = work.getContext().getOptionManager();
     final boolean isUserLoggedIn = AuthDynamicFeature.isUserLoggedIn(sc);
-    String adminUsers = isUserLoggedIn ?
-            ExecConstants.ADMIN_USERS_VALIDATOR.getAdminUsers(optionManager) : null;
-    String adminUserGroups = isUserLoggedIn ?
-            ExecConstants.ADMIN_USER_GROUPS_VALIDATOR.getAdminUserGroups(optionManager) : null;
+    final String processUser = ImpersonationUtil.getProcessUserName();
+    final String processUserGroups = Joiner.on(", ").join(ImpersonationUtil.getProcessUserGroupNames());
+    String adminUsers = ExecConstants.ADMIN_USERS_VALIDATOR.getAdminUsers(optionManager);
+    String adminUserGroups = ExecConstants.ADMIN_USER_GROUPS_VALIDATOR.getAdminUserGroups(optionManager);
 
     // separate groups by comma + space
-    if (adminUsers != null) {
+    if (adminUsers.length() == 0) {
```

Comment:
Frankly saying, I would suggest to do all these manipulations in freemarker. There are many built-in freemarker functions that can help (i.e. split, join, check for emptiness etc [1]).

[1] http://freemarker.org/docs/ref_builtins_string.html#ref_builtin_split

Reply:
@arina-ielchiieva Thanks for the suggestion.
Any particular reason to move these manipulations to freemarker? Do we have any advantage moving this if block over to freemarker?

Reply:
This would be adherence to the MVC pattern which separates model from the view. Model is generated on the server side while freemarker is responsible for representation layer.

Reply:
Made the changes. @arina-ielchiieva please review